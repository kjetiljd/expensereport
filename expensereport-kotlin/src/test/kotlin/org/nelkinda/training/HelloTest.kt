package org.nelkinda.training

import com.github.stefanbirkner.systemlambda.SystemLambda.tapSystemOut
import org.approvaltests.Approvals
import org.junit.Ignore
import org.junit.Test
import kotlin.test.assertEquals
import kotlin.test.assertTrue

class HelloTest {

    @Test
    fun dummy() {

    }

    @Test
    fun `capture print output and make sure it is not empty`() {
        val noExpenses = emptyList<Expense>()

        val actual = tapSystemOut {
            ExpenseReport().printReport(noExpenses)
        }

        assertTrue(actual.isNotEmpty())
    }

    @Test @Ignore("WIP")
    fun `first line in report has correct format`() {
        val noExpenses = emptyList<Expense>()

        val actual = tapSystemOut {
            ExpenseReport().printReport(noExpenses)
        }

        // assert first line in report is correct format

    }

    @Test
    @Ignore("Depends on current time")
    fun `golden master test of program output`() {
        val noExpenses = emptyList<Expense>()

        val actual = tapSystemOut {
            ExpenseReport().printReport(noExpenses)
        }
        Approvals.verify(actual)
    }

}
