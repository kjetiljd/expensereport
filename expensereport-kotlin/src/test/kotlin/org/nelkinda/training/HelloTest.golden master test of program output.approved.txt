Expenses Thu Jan 13 12:40:13 CET 2022
Meal expenses: 0
Total expenses: 0
